src/components/metadata/form/index.js

```diff
@@ -0,0 +1,514 @@
+ * @author psarando
+ */
+import React from "react";
+
+import { FieldArray, Formik } from "formik";
+import { Trans } from "react-i18next";
+import { useMutation, useQuery } from "react-query";
+import PropTypes from "prop-types";
+
+import { useTranslation } from "i18n";
+
+import constants from "../../../constants";
+
+import ids from "../ids";
+
+import AVUFormList from "./AVUFormList";
+import MetadataFormToolbar from "./MetadataFormToolbar";
+import MetadataList from "../listing";
+
+import SaveAsDialog from "components/data/SaveAsDialog";
+import { getParentPath, isWritable } from "components/data/utils";
+
+import ConfirmationDialog from "components/utils/ConfirmationDialog";
+import SignInDialog from "components/utils/SignInDialog";
+import ExternalLink from "components/utils/ExternalLink";
+
+import { ERROR_CODES, getErrorCode } from "components/utils/error/errorCode";
+import withErrorAnnouncer from "components/utils/error/withErrorAnnouncer";
+
+import { useBootstrapInfo } from "contexts/bootstrap";
+import { useUserProfile } from "contexts/userProfile";
+
+import {
+    FILESYSTEM_METADATA_QUERY_KEY,
+    getFilesystemMetadata,
+    saveFilesystemMetadata,
+    setFilesystemMetadata,
+} from "serviceFacades/metadata";
+
+import { AnnouncerConstants, announce, build } from "@cyverse-de/ui-lib";
+
+import { AppBar, Tab, Tabs } from "@material-ui/core";
+
+const MetadataFormListing = (props) => {
+    const {
+        baseId,
+        editable,
+        loading,
+        fetchError,
+        metadata,
+        targetResource,
+        onSelectTemplateBtnSelected,
+        onSaveMetadataToFileBtnSelected,
+        // from formik
+        handleSubmit,
+        resetForm,
+        setValues,
+        setFieldValue,
+        dirty,
+        isSubmitting,
+        errors,
+        values,
+    } = props;
+
+    const { avus, "irods-avus": irodsAVUs } = values;
+    const targetName = targetResource?.label;
+
+    const { t } = useTranslation("metadata");
+
+    const [
+        showImportConfirmationDialog,
+        setShowImportConfirmationDialog,
+    ] = React.useState(false);
+    const [editingAttrIndex, setEditingAttrIndex] = React.useState(-1);
+    const [tabIndex, setTabIndex] = React.useState(0);
+    const [irodsAVUsSelected, setIrodsAVUsSelected] = React.useState([]);
+
+    React.useEffect(() => {
+        // The presenter wants to load metadata for a different target,
+        // usually after already closing this dialog,
+        // so reset the view to default settings.
+        setTabIndex(0);
+        setIrodsAVUsSelected([]);
+
+        resetForm();
+    }, [resetForm, targetResource]);
+
+    React.useEffect(() => {
+        // The presenter wants to load new metadata,
+        // possibly for the same target, and usually from a metadata template,
+        // so reset the view to default settings.
+        setTabIndex(0);
+        setIrodsAVUsSelected([]);
+
+        setValues(metadata);
+    }, [setValues, metadata]);
```

Comment:
Since this listing is no longer intended for a dialog, I'll have to test this once I integrate the listing in a next.js page.
But that reminds me that I need to update the comments for the `targetResource` effect 😅